shared/util/preconditions.go

```diff
@@ -12,17 +12,61 @@ import (
 	"github.com/pkg/errors"
 )
 
-func ValidateEnvironment(ctx context.Context) error {
-	dockerComposeMinVersion, err := semver.NewConstraint(">= v2")
-	if err != nil {
-		return errors.Wrap(err, "could not establish docker compose version")
+type ValidationCallback = func(context.Context) error
+
+type ValidationCheckList struct {
+	MinDockerComposeVersion          ValidationCallback
+	DockerInstalled                  ValidationCallback
+	DockerEngineRunning              ValidationCallback
+	AwsInstalled                     ValidationCallback
+	TerraformInstalled               ValidationCallback
+	AwsSessionManagerPluginInstalled ValidationCallback
+}
```

Comment:
I feel this struct is a bit unnecessary, but I have very weak feelings about it

Reply:
Can this be a map?

Reply:
It can be anything, or nothing. The idea was to "bundle" them up to make code completion work better. Otherwise it's just a bunch of functions in the util namespace. There's nothing special about it otherwise.